File: src/main/java/net/serebryansky/common/service/RoutingService.java
package net.serebryansky.common.service;

import net.serebryansky.place.model.Place;
import net.serebryansky.routing.model.DurationMatrixResponse;
import net.serebryansky.routing.model.Route;
import org.reactivestreams.Publisher;
import reactor.core.publisher.Mono;

import java.util.List;

public interface RoutingService {
    Mono<DurationMatrixResponse> getDurationMatrix(List<Place> places);

    Mono<Route> getRoute(List<Place> places);
}

File: src/main/kotlin/net/serebryansky/common/service/RoutingService.kt
package net.serebryansky.common.service

import net.serebryansky.place.model.Place
import net.serebryansky.routing.model.DurationMatrixResponse
import net.serebryansky.routing.model.Route
import reactor.core.publisher.Mono

interface RoutingService {
    suspend fun getDurationMatrix(places: List<Place>): DurationMatrixResponse
    suspend fun getRoute(places: List<Place>): Route
}
